APIWrapper/Proxy.cs

```diff
@@ -0,0 +1,123 @@
+        {
+            string address = $"{__URL__}/{__SCHEDULE_CONTENT_URL__}?m={type}&id={Uri.EscapeUriString(requestContent)}";
+
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(address);
+            request.Method = "Get";
+            request.KeepAlive = true;
+            request.ContentType = "application/x-www-form-urlencoded";
+
+            HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+            string myResponse = "";
+            using (System.IO.StreamReader sr = new System.IO.StreamReader(response.GetResponseStream()))
+            {
+                myResponse = sr.ReadToEnd();
+            }
+            
+            return ParseResponse(myResponse);
+        }
+
+        private ScheduleWeekContent ParseResponse(string myResponse)
+        {
+            
+            JObject joResponse = JObject.Parse(myResponse);
+            //check if error occured
+            string report = joResponse["report"].ToString();
+            if (!String.IsNullOrWhiteSpace(report))
+            {
+                throw new ArgumentException(report);
+            }
+            
+            JArray array = (JArray)joResponse["ScheduleContent"];
+            ScheduleWeekContent ret = new ScheduleWeekContent();
+            foreach (var element in array)
+            {
+                int blockNumber = 1;
+                var daySchedule = new ScheduleDayContent();
+                foreach (var blck in element)
+                {
+                    ScheduleHourContent sc = null;
+                    try
+                    {
+                        sc = new ScheduleHourContent();
+                        sc.BlockNumber = blockNumber++;
+                        sc.IsBlocked = Convert.ToBoolean(int.Parse(blck["b"].ToString()));
+
+                        if (!String.IsNullOrWhiteSpace(blck["t"].ToString()) && !String.IsNullOrWhiteSpace(blck["p"].ToString()))
+                        {
+                            sc.LessonType = this.ConvertLessonType(blck["t"].ToString()[0]);
+                            sc.TeacherName = blck["u"].ToString();
+                            sc.RoomName = blck["r"].ToString();
+                            sc.SubjectShortcut = blck["s"].ToString();
+                            sc.SubjectName = blck["k"].ToString();
+                            sc.StudyGroups = blck["g"].ToString().Split(',').Select(x => x.Trim()).ToList();
+                            sc.SubjectType = (ESubjectType)Convert.ToInt32(blck["p"].ToString());
+                            sc.IsEmptyBlock = false;
+                        }
+                        else
+                        {
+                            sc.IsEmptyBlock = true;
+                        }
+                        daySchedule.BlocksInDay.Add(sc);
+                    }
+                    catch (Exception ex)
+                    {
+                        //to do logger
```

Comment:
Prosim toto nikdy..., aj ked to aspon pises na konzolu... ale minimalne by som to dal do logu a sucasne vyhodil vynimku vyssie... (vlastnu a s popisnym textom)